sys/cddl/dev/dtrace/dtrace_ioctl.c

```diff
@@ -214,32 +248,51 @@ dtrace_ioctl(struct cdev *dev, u_long cmd, caddr_t addr,
 		return (0);
 	}
 	case DTRACEIOC_AGGSNAP:
-	case DTRACEIOC_BUFSNAP: {
-		dtrace_bufdesc_t **pdesc = (dtrace_bufdesc_t **) addr;
-		dtrace_bufdesc_t desc;
+	case DTRACEIOC_BUFSNAP:
+	case DTRACEIOC_BUFSNAP_64: {
+		dtrace_bufdesc_t *__capability *pdesc =
+		    (dtrace_bufdesc_t * __capability *)addr;
+		union dtrace_bufdesc_union desc;
 		caddr_t cached;
 		dtrace_buffer_t *buf;
 
 		dtrace_debug_output();
 
-		if (copyin((void *) *pdesc, &desc, sizeof (desc)) != 0)
+		dtrace_bufdesc_t *__capability pdesc_cap = SV_CURPROC_FLAG(SV_CHERI) ?
```

Comment:
May be cleaner to have a separate `copyin_bufdesc()` that fills a given `bufdesc_union`, a similar thing is also valid for other copy-in/outs.

Reply:
Done it (: